### PR TITLE
Ests 136457

### PR DESCRIPTION
**Pull request by silvam11**

### CODE REVIEW

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/AddVCenterHostToProtectionDomain.java [hunk 1]
```diff
@@ -112,85 +140,129 @@ else if (job.getInputParams().getEsxiManagementIpAddress() != null){
...
-             *//*
-            final NodeExpansionRequest inputParams = job.getInputParams();
-            final String protectionDomain = inputParams.getProtectionDomain();
+        final Map<String, DeviceAssignment> deviceToDeviceStoragePoolAssignment = nodeDetail.getDeviceToDeviceStoragePool();
 
-            if (protectionDomain == null)
-            {
-                throw new IllegalStateException("No Protection domain provided");
-            }
+        if (protectionDomainId == null)
```
Comment (aroberson) [inline dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/AddVCenterHostToProtectionDomain.java]: This is not needed.  The protection domain id should always be there based on first step of workflow.

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/UpdatePCIPassThrough.java [hunk 2]
```diff
@@ -6,26 +6,37 @@
...
+import org.springframework.util.StringUtils;
 
+import static com.dell.cpsd.paqx.dne.service.delegates.utils.DelegateConstants.DEPLOY_SCALEIO_NEW_VM_NAME;
+import static com.dell.cpsd.paqx.dne.service.delegates.utils.DelegateConstants.HOSTNAME;
+import static com.dell.cpsd.paqx.dne.service.delegates.utils.DelegateConstants.HOST_PCI_DEVICE_ID;
 import static com.dell.cpsd.paqx.dne.service.delegates.utils.DelegateConstants.NODE_DETAIL;
+import static com.dell.cpsd.paqx.dne.service.delegates.utils.DelegateConstants.UPDATE_PCI_PASSTHROUGH;
 
+/**
+ * Set PCI IP pass through.
```
Comment (kushagra-mahajan) [inline dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/UpdatePCIPassThrough.java]: Update PCI Passthrough Controller for ScaleIO VM?

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/DeployScaleIOVm.java [hunk 3]
```diff
@@ -78,107 +103,119 @@ public DeployScaleIOVm(final NodeService nodeService, final DataServiceRepositor
...
+        }
+
+        delegateExecution.setVariable(DEPLOY_SCALEIO_NEW_VM_NAME, newScaleIoVmName);
+
+        String returnMessage = taskMessage + " on Node " + nodeDetail.getServiceTag() + " was successful.";
+        LOGGER.info(returnMessage);
+        updateDelegateStatus(returnMessage);
     }
 
     private class Validate
```
Comment (aroberson) [inline dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/DeployScaleIOVm.java]: Do we still need this?  The fields should have already been validated before getting to this step?  Seems like not necessary.

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/UpdatePCIPassThrough.java [hunk 4]
```diff
@@ -70,105 +81,79 @@ public void delegateExecute(final DelegateExecution delegateExecution)
...
+        }
 
-        LOGGER.info(taskMessage + " on Node " + nodeDetail.getServiceTag() + " was successful.");
-        updateDelegateStatus(taskMessage + " on Node " + nodeDetail.getServiceTag() + " was successful.");
+        String returnMessage = taskMessage + " on Node " + nodeDetail.getServiceTag() + " was successful.";
+        LOGGER.info(returnMessage);
+        updateDelegateStatus(returnMessage);
     }
 
     private class Validate
```
Comment (aroberson) [inline dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/UpdatePCIPassThrough.java]: Not sure this is needed either.  You would not get to this step in workflow without those being set.

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/AddVCenterHostToProtectionDomain.java (hunk 1 above)
Comment (silvam11) [reply dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/AddVCenterHostToProtectionDomain.java]: done

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/DeployScaleIOVm.java (hunk 3 above)
Comment (silvam11) [reply dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/DeployScaleIOVm.java]: done

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/UpdatePCIPassThrough.java (hunk 4 above)
Comment (silvam11) [reply dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/UpdatePCIPassThrough.java]: done

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/UpdatePCIPassThrough.java (hunk 2 above)
Comment (silvam11) [reply dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/delegates/UpdatePCIPassThrough.java]: done

**Pull request merged**